server/src/main/java/org/elasticsearch/action/admin/cluster/stats/CCSUsage.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.action.admin.cluster.stats;
+
+import org.elasticsearch.core.TimeValue;
+
+import java.util.Map;
+
+/**
+ * This is a snapshot of telemetry from an individual cross-cluster search for _search or _async_search (or
+ * other search endpoints that use the TransportSearchAction such as _msearch).
+ */
+public class CCSUsage {
+    private final long took;
+    private final String failureType;  // TODO: enum?
+    private final boolean minimizeRoundTrips;
+    private final boolean async;
+    private final int skippedRemotes;
+    private final Map<String, PerClusterUsage> perClusterUsage;
+
+    public static class Builder {
```

Comment:
What is the function of the builder? Couldn't we just call a ctor? Is it a readability function?